astropy/utils/xml/validate.py

```diff
@@ -47,10 +48,8 @@ def validate_schema(filename, schema_file):
     if p.returncode == 127:
         raise OSError("xmllint not found, so can not validate schema")
     elif p.returncode < 0:
-        from astropy.utils.misc import signal_number_to_name
-
         raise OSError(
-            f"xmllint was terminated by signal '{signal_number_to_name(-p.returncode)}'"
+            f"xmllint was terminated by signal '{Signals(-p.returncode).name}'"
```

Comment:
This does change function behavior in a subtle way for people who use XML validator (like `astroquery.vo_conesearch`), so I think it needs a change log.

Reply:
What changes are you talking about?

Reply:
In your post, you mentioned, "There is a difference regarding invalid signal numbers. `signal_number_to_name()` returns `'UNKNOWN'`, whereas `Signals` raises an error. "

Reply:
I can't imagine the OS actually sending an invalid signal, other than a cosmic ray hitting the CPU and causing a bit flip, but that could cause any code to crash. I mentioned it because the code path for an invalid signals exists in the `signal_number_to_name()` source code, but I really don't think it's relevant at all in practice, which is why I didn't bother replacing it.

Reply:
Makes sense. Thanks!